crates/cheatcodes/src/evm/fork.rs

```diff
@@ -355,3 +362,13 @@ fn check_broadcast(state: &Cheatcodes) -> Result<()> {
         Err(fmt_err!("cannot select forks during a broadcast"))
     }
 }
+
+// Helper to ensure that the caller managing the fork is persistent.
+// Applies to create, select and roll forks actions.
+// https://github.com/foundry-rs/foundry/issues/8004
+#[inline]
+fn ensure_persistent_caller<DB: DatabaseExt>(ccx: &mut CheatsCtxt<DB>) {
```

Comment:
I think we can replace this function entirely and always call add which is more efficient because it only does one lookup

but I can see the value in this function for doc purposes, but would like to drop the `ensure_` prefix because this is commonly used by functions that check a condition and return an error if that condition is violated which is not the case here

Reply:
changed with https://github.com/foundry-rs/foundry/pull/8041/commits/4a6fae0ee51a3bd6b0281b0832f25f58f86f8869